### PR TITLE
Deflake pod-zone-affinity integration test

### LINKED ISSUE #6648: [Flaky Test] `PodZoneAffinity` Integration Test Suite
**How to categorize this issue?**

/area testing
/kind flake

**Which test(s)/suite(s) are flaking**:
`PodZoneAffinity` Integration Test Suite

**CI link**:
- https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/6647/pull-gardener-integration/1567724664074014720

**Reason for failure**:
```
• [FAILED] [0.014 seconds]
PodSchedulerName tests
/home/prow/go/src/github.com/gardener/gardener/test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_test.go:26
  when namespace has zone enforcement label with value
  /home/prow/go/src/github.com/gardener/gardener/test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_test.go:81
    [It] should add nodeAffinity
    /home/prow/go/src/github.com/gardener/gardener/test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_test.go:96
  Begin Captured GinkgoWriter Output >>
    {"level":"debug","ts":"2022-09-08T04:04:05.188Z","logger":"controller-runtime.webhook.webhooks","msg":"received request","webhook":"/webhooks/pod-zone-affinity","UID":"b2cd2da0-b6c1-4c5f-aa06-baf99d7c5c60","kind":"/v1, Kind=Pod","resource":{"group":"","version":"v1","resource":"pods"}}
    {"level":"debug","ts":"2022-09-08T04:04:05.188Z","logger":"controller-runtime.webhook.webhooks","msg":"wrote response","webhook":"/webhooks/pod-zone-affinity","code":200,"reason":"","UID":"b2cd2da0-b6c1-4c5f-aa06-baf99d7c5c60","allowed":true}
  << End Captured GinkgoWriter Output
  Expected
      <*v1.NodeAffinity | 0x0>: nil
  not to be nil
  In [It] at: /home/prow/go/src/github.com/gardener/gardener/test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_test.go:101
------------------------------
•
Ran 3 of 3 Specs in 6.686 seconds
FAIL! -- 2 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestPodPodZoneAffinity (6.69s)
FAIL
FAIL	github.com/gardener/gardener/test/integration/resourcemanager/podzoneaffinity	6.799s
```

**Anything else we need to know**:
Introduced with #6579 and first seen in https://github.com/gardener/gardener/pull/6647#issuecomment-1240193626

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area testing
/kind flake

**What this PR does / why we need it**:
This PR fixes flaky integration tests for pod-zone-affinity, reported in #6648.

**Which issue(s) this PR fixes**:
Fixes #6648

**Special notes for your reviewer**:
The main issue was that `BeforeEach` steps removed the test-id label (`corev1.LabelMetadataName`) that was used in the `MutatingWebhookConfiguration` as a namespace selector. However, after iterating over the tests again I decided to improve additional points, in sum:
- Only add and remove `control-plane.shoot.gardener.cloud/enforce-zone` label -> solves above depicted problem
- Remove namespace name label selector from `MutatingWebhookConfiguration` since we already use another label selector which is sufficient. Also, the `corev1.LabelMetadataName` caused problems when `stress` testing.
- Disable client cache for `Namespace` resources, so that modifications in `BeforeEach` are immediately seed by the manager client.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_suite_test.go [hunk 1]
```diff
@@ -157,11 +161,5 @@ func getMutatingWebhookConfigurations(namespace string) []*admissionregistration
 		},
 	}
 
-	webhookConfig[0].Webhooks[0].NamespaceSelector.MatchExpressions = append(webhookConfig[0].Webhooks[0].NamespaceSelector.MatchExpressions, metav1.LabelSelectorRequirement{
-		Key:      corev1.LabelMetadataName,
-		Operator: metav1.LabelSelectorOpIn,
-		Values:   []string{namespace},
-	})
```
Comment (rfranzke) [inline test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_suite_test.go:164]: Why did you remove this? 👀

Comment (timuthy) [reply test/integration/resourcemanager/podzoneaffinity/podzoneaffinity_suite_test.go:164]: Please see PR description:
> Remove namespace name label selector from MutatingWebhookConfiguration since we already use another label selector which is sufficient. Also, the corev1.LabelMetadataName caused problems when stress testing.

If you use stress testing with a single `MutatingWebhookConfiguration` then you have to expand the namespace selector everytime a namespace is created which will result in conflicts. I think the other namespace selector `"control-plane.shoot.gardener.cloud/enforce-zone"` (already in place) is already sufficient for our integration tests as it makes sure that non-related namespaces are excluded.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve
